e2e/container/create_test.go

```diff
@@ -32,4 +38,69 @@ func TestCreateWithContentTrust(t *testing.T) {
 	})
 }
 
-// FIXME(vdemeester) TestTrustedCreateFromBadTrustServer needs to be backport too (see https://github.com/moby/moby/pull/36515/files#diff-4b1e56bb77ac16f2ccf956fc24cf0a82L331)
+func TestTrustedCreateFromUnreachableTrustServer(t *testing.T) {
+	dir := fixtures.SetupConfigFile(t)
+	defer dir.Remove()
+	image := fixtures.CreateMaskedTrustedRemoteImage(t, registryPrefix, "trust-create", "latest")
+
+	result := icmd.RunCmd(
+		icmd.Command("docker", "create", image),
+		fixtures.WithConfig(dir.Path()),
+		fixtures.WithTrust,
+		fixtures.WithNotaryServer("https://invalidnotaryserver"),
+	)
+	result.Assert(t, icmd.Expected{
+		ExitCode: 1,
+		Err:      "error contacting notary server",
+	})
+}
+
+func TestTrustedCreateFromBadTrustServer(t *testing.T) {
+	dir := fixtures.SetupConfigFile(t)
+	defer dir.Remove()
+
+	evilNotaryDir := setupEvilNotaryServer(t)
+	defer evilNotaryDir.Remove()
+
+	icmd.RunCmd(icmd.Command("docker", "pull", fixtures.AlpineImage)).Assert(t, icmd.Success)
+	icmd.RunCmd(icmd.Command("docker", "tag", fixtures.AlpineImage, evilImageName),
+		fixtures.WithConfig(evilNotaryDir.Path()),
+	).Assert(t, icmd.Success)
+
+	icmd.RunCmd(icmd.Command("docker", "image", "push", evilImageName),
+		fixtures.WithConfig(evilNotaryDir.Path()),
+		fixtures.WithPassphrase("root_password", "repo_password"),
+		fixtures.WithTrust,
+		fixtures.WithNotaryServer(evilNotaryURL),
+	).Assert(t, icmd.Success)
+
+	icmd.RunCmd(icmd.Command("docker", "image", "rm", evilImageName)).Assert(t, icmd.Success)
+
+	icmd.RunCmd(icmd.Command("docker", "create", evilImageName),
+		fixtures.WithConfig(dir.Path()),
+		fixtures.WithTrust,
+		fixtures.WithNotary,
+	).Assert(t, icmd.Expected{
+		ExitCode: 1,
+		Err:      "remote trust data does not exist for registry:5000/evil-alpine",
+	})
+
+}
+
+func setupEvilNotaryServer(t *testing.T) fs.Dir {
+	t.Helper()
+	dir := fs.NewDir(t, "evil_notary_test", fs.WithMode(0700), fs.WithFile("config.json", `
+	{
+		"auths": {
+			"registry:5000": {
+				"auth": "ZWlhaXM6cGFzc3dvcmQK"
+			},
+			"https://evil-notary-server:4444": {
```

Comment:
This seems to be the only difference from fixtures.SetupConfigFile, maybe pass notartURL as an argument there?

Reply:
yes this is the only difference. I am going to update the code.